Source/Engine/Physics/Joints/Joint.cpp

```diff
@@ -224,15 +281,90 @@ void Joint::DrawPhysicsDebug(RenderView& view)
-        DEBUG_DRAW_WIRE_SPHERE(BoundingSphere(GetTargetPosition(), 4.0f), Color::AliceBlue * 0.8f, 0, true);
+        PhysicsTransform B{};
+        PhysicsTransform A{};
+        if (_joint) {
+            PhysicsBackend::GetJointLocalPose(_joint, 0, A);
+            PhysicsBackend::GetJointLocalPose(_joint, 1, B);
+        }
+        //cashe for cheak sync on A
+        PhysicsTransform& fA = LocalPoseActor0;
+        PhysicsTransform& pxA = A;
+        //cashe for cheak sync on B
+        PhysicsTransform& fB = LocalPoseActor1;
+        PhysicsTransform& pxB = B;
+
+        bool WarnA = false;
+        bool WarnB = false;
+
+        if (fA.Translation != pxA.Translation || fA.Orientation != pxA.Orientation)
+        {
+            WarnA = true;
+        }
+        if (fB.Translation != pxB.Translation || fB.Orientation != pxB.Orientation)
+        {
+            WarnA = true;
+        }
+
+        if (Actor0 != nullptr) 
+        {
+            A = PhysicsTransform::LocalToWorld(Actor0->GetTransform(), A);
+        }
+
+        if (Actor1 != nullptr) 
+        {
+            B = PhysicsTransform::LocalToWorld(Actor1->GetTransform(), B);
+        }
+
+        auto p = Vector3::Lerp(A.Translation, B.Translation, 0.5f);
+        bool d = Vector3::Distance(view.WorldPosition, p) > 500.0f;
+
+        DEBUG_DRAW_WIRE_SPHERE(BoundingSphere(A.Translation, 3.0f), Color::Red * 0.8f, 0, d);
+        DEBUG_DRAW_LINE(A.Translation, B.Translation, Color::Aqua, 0, d);
+        DEBUG_DRAW_WIRE_SPHERE(BoundingSphere(B.Translation, 4.0f), Color::Blue * 0.8f, 0, d);
+        if (d)
+        {
+            if (!WarnA && !WarnB)
+                return;
+
+            String WarnMsg{ TEXT("Warning joint is not in sync with back end\n") };
+            if (WarnA)
+            {
+                WarnMsg.Append(TEXT("A is off sync\nValues:\n"));
+                WarnMsg.Append(A.Translation.ToString());
+                WarnMsg.Append(TEXT("\n"));
+                WarnMsg.Append(A.Orientation.GetEuler().ToString());
+                WarnMsg.Append(TEXT("\n"));
+            }
+            if (WarnB)
+            {
+                WarnMsg.Append(TEXT("B is off sync\n"));
+
+                WarnMsg.Append(B.Translation.ToString());
+                WarnMsg.Append(TEXT("\n"));
+                WarnMsg.Append(B.Orientation.GetEuler().ToString());
+                WarnMsg.Append(TEXT("\n"));
+            }
+            DEBUG_DRAW_TEXT(WarnMsg, GetPosition(), Color::Yellow, 5, 0);
+
+        }
     }
 }
 
 void Joint::OnDebugDrawSelected()
 {
-    DEBUG_DRAW_WIRE_SPHERE(BoundingSphere(GetPosition(), 3.0f), Color::BlueViolet * 0.8f, 0, false);
-    DEBUG_DRAW_WIRE_SPHERE(BoundingSphere(GetTargetPosition(), 4.0f), Color::AliceBlue * 0.8f, 0, false);
+    auto p0 = GetPoseActor0();
+    auto p1 = GetPoseActor1();
+    if (Actor0) {
+        DEBUG_DRAW_WIRE_BOX(Actor0->GetBox(), Color::BlueViolet * 0.8f, 0, false);
+    }
```

Comment:
Move starting bracket to new line for code consistency, or since it is only a single line of code after, remove brackets.